### ISSUE TITLE
Problems while syncing with Nextcloud on iOS/macOS

### ISSUE THREAD

**Issue by jcgerhard** [laurent22/joplin#218]
----
### Operating system

- [ ] Windows
- [x] macOS
- [ ] Linux
- [ ] Android
- [x] iOS

### Application

- [x] Desktop
- [x] Mobile
- [ ] Terminal
----
#### macOS

![joplin](https://user-images.githubusercontent.com/29855366/36244847-4c20815c-1228-11e8-9355-2525868092f1.jpg)

#### iOS

![app](https://user-images.githubusercontent.com/29855366/36245263-9641dacc-122a-11e8-8137-0338e301989b.jpg)

----

I have this strange issue while syncing Joplin with Nextcloud.
The first step I made, was setting up the mobile app on iOS.
As one can see here, the credentials are working fine and syncing runs flawless.

![ios](https://user-images.githubusercontent.com/29855366/36244349-2aec240c-1226-11e8-82b4-2a7510aaed3e.jpg)

I checked the regarding target folder in the Nextcloud instance. All of the files were transferred successfully (incl. images).

----

Then I installed Jolpin on macOS, took the exactly same credentials and got this error. I tried it a lot of times but can't get it work properly.

![macos](https://user-images.githubusercontent.com/29855366/36244432-7481c608-1226-11e8-97d8-32d085a7f584.jpg)

The log shows the following informations:
```
2018-02-15 08:20:13: "Profile directory: /Users/jangerhard/.config/joplin-desktop"
2018-02-15 08:20:13: "Trying to load 0 master keys..."
2018-02-15 08:20:13: "Loaded master keys: 0"
2018-02-15 08:20:13: "Trying to load 0 master keys..."
2018-02-15 08:20:13: "Loaded master keys: 0"
2018-02-15 08:20:13: "Scheduling sync operation..."
2018-02-15 08:20:13: "Scheduling sync operation..."
2018-02-15 08:20:14: "DecryptionWorker: starting decryption..."
2018-02-15 08:20:14: "DecryptionWorker: completed decryption."
2018-02-15 08:20:43: "Preparing scheduled sync"
2018-02-15 08:20:43: "Starting scheduled sync"
2018-02-15 08:21:05: "Error: PROPFIND : <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>411 Length Required</title>
</head><body>
<h1>Length Required</h1>
<p>A request of the requested method PROPFIND requires a valid Content-length.<br />
</p>
</body></html>

Code: 411
Error: PROPFIND : <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>411 Length Required</title>
</head><body>
<h1>Length Required</h1>
<p>A request of the requested method PROPFIND requires a valid Content-length.<br />
</p>
</body></html>

    at WebDavApi.exec (/Applications/Joplin.app/Contents/Resources/app/lib/WebDavApi.js:287:10)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:109:7)"
2018-02-15 08:21:05: "Operations completed: "
2018-02-15 08:21:05: "Total folders: 0"
2018-02-15 08:21:05: "Total notes: 0"
2018-02-15 08:21:05: "Total resources: 0"
2018-02-15 08:21:05: "There was some errors:"
2018-02-15 08:21:05: "Error: PROPFIND : <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>411 Length Required</title>
</head><body>
<h1>Length Required</h1>
<p>A request of the requested method PROPFIND requires a valid Content-length.<br />
</p>
</body></html>

Code: 411
Error: PROPFIND : <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>411 Length Required</title>
</head><body>
<h1>Length Required</h1>
<p>A request of the requested method PROPFIND requires a valid Content-length.<br />
</p>
</body></html>

    at WebDavApi.exec (/Applications/Joplin.app/Contents/Resources/app/lib/WebDavApi.js:287:10)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:109:7)"
2018-02-15 08:21:05: "Updading all notifications..."
2018-02-15 08:21:05: "Garbage collecting alarms..."
2018-02-15 08:21:06: "DecryptionWorker: starting decryption..."
2018-02-15 08:21:06: "DecryptionWorker: completed decryption."
```
----

I also changed from target "Nextcloud" to target "WebDAV" and also used the alternative path:
```
.../remote.php/dav/username/files/Backups/Joplin
```
Maybe I did something wrong? Maybe it's a bug? Maybe someone can help?

----
**By the way:** PLEASE enable clipboard copy + paste in the settings window as mentioned here: #46

**Comment by laurent22** [comment]
Out of curiosity on which server is your Nextcloud hosted (which version of Apache or Nginx)? My understanding is that with HTTP/1.0, the Content-Length header is required, while with HTTP/1.1 it is only recommended (but can be omitted). Currently Joplin doesn't send this header at all so that's why you are seeing this error.

A quick fix on your side would be to tweak your server so that it allows requested with no Content-Length header, maybe by switching it to HTTP/1.1. Eventually I will also make the app send this Content-Length header anyway.

I think it works in iOS because the data is transferred in chunks in which case the Content-Length is not required.

**Comment by jcgerhard** [comment]
I am hosting my Nextcloud at a external provider. It's an apache server that uses the HTTP/1.1 version. All changes I have to realize with the .htaccess file. Is there an option for that? Maybe it would be a good idea to make joplin sending the content-length information?

**Comment by laurent22** [comment]
Yes, in the next version it will send the content-length header. I had this temporarily disabled due to issues with non-English text but this should be fixed now.

**Comment by jcgerhard** [comment]
This would be absolutely amazing! I really like Joplin and meanwhile I use it every day on macOS and iOS but can‘t really keep in step. And I think it would allow more people to use it regularly with a shared hosting or providers like mine who doesn‘t provide the opportunity to change such apache or nginx configurations. Thanks in advance!

**Comment by laurent22** [comment]
Fixed in 3a4bbd571e5608bb8cd81772738d9ccc9dffb419

**Comment by jcgerhard** [comment]
Thank you for fixing this issue! Synchronization between macOS, Nextcloud and iOS perfectly works now! And by sending the content-length header now you fixed another bug I recognized this morning before upgrading to the latest version: on macOS, attached images aren't synchronized. This works flawlessly now, too! Many thanks for your dedication! Joplin is a wonderful handy application!
